### ISSUE TITLE
Support for FileJump?

### ISSUE THREAD

**Issue by closewall** [rclone/rclone#8114]
Will rclone add support for FileJump? It appears to have an open API with supplied documentation: https://drive.filejump.com/api-docs

**Comment by albertony** [comment]
I wouldn't hold my breath.. 🙂 This is the 81. open issue regarding support for a new backend. Unless you plan to implement it yourself, or sponsor it making it more interesting for someone else to invest time on it, it is unlikely to happen any time soon. Sorry.

**Comment by masrlinu** [comment]
I am on my way to add support for FileJump ☺️

**Comment by masrlinu** [comment]
Listing a directory already almost works, but I still have problems. So whoever wants to help:

https://forum.rclone.org/t/help-with-adding-new-backend-filejump-com/48127

But by the way, ftp (which is supported by rclone) is on the roadmap:
https://drive.filejump.com/pages/roadmap
I asked the filejump support. They answered:
"It's under testing phase shall be released soon"

**Comment by masrlinu** [comment]
We are so close to supporting the new backend FileJump 😊 But I still need some help!!

Listing the folder works now, for example with
```
./rclone lsjson filejump:
```
I implemented all methods of the backend, but there are still some errors. It would be good, if somebody with experience could take a look on them and maybe correct my [code](https://github.com/masrlinu/rclone).

Here is this [issue in the rclone-forum](https://forum.rclone.org/t/help-we-are-so-close-to-supporting-the-new-backend-filejump-but-i-need-some-help/48244).

By the way, if you want to check the Api, you can just go to [filejump.com](https://filejump.com) and check the network connections in the developer tools, while you do something. The requests are very easy to understand.

**Comment by masrlinu** [comment]
Syncing via
```
./rclone sync /tmp/tosync/ filejump:/tosync
```
works now, but the files get copied twice, I still have to find out, why.

**Comment by ncw** [comment]
@masrlinu are you running the integration tests? If you can make the integration tests pass then the backend will work perfectly. The way I develop a new backend is just to plod through the integration tests making them all pass.

Make [these ones work first](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#unit-tests) and then [the full tests](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#integration-tests)

Also take a look at the other things in the writing a new backend docs if you haven't already.

**Comment by GhsVilela** [comment]
> Syncing via
> 
> ```
> ./rclone sync /tmp/tosync/ filejump:/tosync
> ```
> 
> works now, but the files get copied twice, I still have to find out, why.

Hey, thanks for you work with this, any updates? I'm interested in filejump support as well

**Comment by masrlinu** [comment]
> @masrlinu are you running the integration tests? 
Thank you, ncw, I started the integration tests like you told me. I got many errors in my code.

> Hey, thanks for you work with this, any updates?
Hi @GhsVilela, FileJump told me that they stopped programming ftp support, because there were errors. But they are programming WebDAV support. You see it on their [Roadmap](https://drive.filejump.com/pages/roadmap).

Rclone supports WebDAV, so as programming Rclone FileJump support is not my full-time job, but programming FileJump WebDAV support is FileJump's full-time job, I think, it is better to wait. I hope, the problem will solve itself soon ☺️

**Comment by addy005** [comment]
@masrlinu anyway to upload big files in filejump? without browser, i have one bot which support rclone uploading anyway i can use filejump there?

**Comment by masrlinu** [comment]
Wenn ich ein filejump: backend in der Config erstelle und das Backend mit
go test -v ./backend/filejump/ -remote filejump:
teste, erhalte ich bei der [aktuellen Version](https://github.com/masrlinu/rclone/tree/f4d327cbcd8176c8aca1e0bb6747d43b07e9d4ae) folgendes Ergebnis:

=== RUN   TestIntegration/FsShutdown
    fstests.go:2706: Shutdown method not supported
--- FAIL: TestIntegration (377.68s)
    --- SKIP: TestIntegration/FsCheckWrap (0.00s)
    --- SKIP: TestIntegration/FsCommand (0.00s)
    --- PASS: TestIntegration/FsRmdirNotFound (0.21s)
    --- PASS: TestIntegration/FsString (0.00s)
    --- PASS: TestIntegration/FsName (0.00s)
    --- PASS: TestIntegration/FsRoot (0.00s)
    --- PASS: TestIntegration/FsRmdirEmpty (0.21s)
    --- FAIL: TestIntegration/FsMkdir (375.60s)
        --- PASS: TestIntegration/FsMkdir/FsMkdirRmdirSubdir (2.61s)
        --- PASS: TestIntegration/FsMkdir/FsListEmpty (0.22s)
        --- PASS: TestIntegration/FsMkdir/FsListDirEmpty (0.22s)
        --- SKIP: TestIntegration/FsMkdir/FsListRDirEmpty (0.00s)
        --- PASS: TestIntegration/FsMkdir/FsListDirNotFound (0.22s)
        --- SKIP: TestIntegration/FsMkdir/FsListRDirNotFound (0.00s)
        --- FAIL: TestIntegration/FsMkdir/FsEncoding (345.61s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/control_chars (9.81s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/dot (0.82s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/dot_dot (0.85s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/punctuation (9.27s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/leading_space (11.80s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/leading_tilde (25.20s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/leading_CR (24.96s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/leading_LF (24.98s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/leading_HT (25.04s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/leading_VT (24.94s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/leading_dot (24.92s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/trailing_space (11.81s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/trailing_CR (25.40s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/trailing_LF (25.17s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/trailing_HT (25.03s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/trailing_VT (25.17s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/trailing_dot (25.14s)
            --- SKIP: TestIntegration/FsMkdir/FsEncoding/invalid_UTF-8 (0.00s)
            --- FAIL: TestIntegration/FsMkdir/FsEncoding/URL_encoding (25.10s)
        --- PASS: TestIntegration/FsMkdir/FsNewObjectNotFound (0.42s)
        --- PASS: TestIntegration/FsMkdir/FsPutError (1.09s)
        --- FAIL: TestIntegration/FsMkdir/FsPutZeroLength (6.29s)
        --- SKIP: TestIntegration/FsMkdir/FsOpenWriterAt (0.00s)
        --- SKIP: TestIntegration/FsMkdir/FsOpenChunkWriter (0.00s)
        --- SKIP: TestIntegration/FsMkdir/FsChangeNotify (0.00s)
        --- FAIL: TestIntegration/FsMkdir/FsPutFiles (6.65s)
        --- SKIP: TestIntegration/FsMkdir/FsPutChunked (0.00s)
        --- SKIP: TestIntegration/FsMkdir/FsCopyChunked (0.00s)
        --- FAIL: TestIntegration/FsMkdir/FsUploadUnknownSize (6.39s)
            --- PASS: TestIntegration/FsMkdir/FsUploadUnknownSize/FsPutUnknownSize (0.00s)
            --- FAIL: TestIntegration/FsMkdir/FsUploadUnknownSize/FsUpdateUnknownSize (6.39s)
        --- PASS: TestIntegration/FsMkdir/FsRootCollapse (1.19s)
        --- SKIP: TestIntegration/FsMkdir/FsDirSetModTime (0.00s)
        --- SKIP: TestIntegration/FsMkdir/FsMkdirMetadata (0.00s)
        --- SKIP: TestIntegration/FsMkdir/FsDirectory (0.00s)
    --- SKIP: TestIntegration/FsShutdown (0.00s)


TestIntegration/FsMkdir/FsEncoding klappt irgendwie garnicht.

Die folgenden drei Tests klappen auch nicht:
TestIntegration/FsMkdir/FsPutZeroLength
TestIntegration/FsMkdir/FsPutFiles
TestIntegration/FsMkdir/FsUploadUnknownSize/FsPutUnknownSize

**Comment by masrlinu** [comment]
I solved all errors! The unit test and the two integration tests passed.
@ncw, if you want to test again, you can use the following rclone.conf:

[filejump]
type = filejump
access_token = 339|ceTxTvuB4l2RceqgAkDlqnQEeFe9KPDO4jeOtFgbe223fbb8
workspace_id = 0

**Comment by framingnoone** [comment]
You should probably not include things like 'access_token' in a bug tracker, for security reasons.

**Comment by masrlinu** [comment]
This is just a test account. I don't store personal data there ☺️

**Comment by masrlinu** [comment]
Filejump has just made a major change to its cloud storage. Everyone can now migrate to the new server by registering with the same email address they used on the old one. They’ve also enabled WebDAV! Unfortunately, Filejump doesn’t provide the token in Settings that I need for my own Filejump backend. I also haven’t tested my backend with the current server yet, so I’m considering deleting it now that WebDAV works — although WebDAV only works for paying customers (lifetime accounts have it for free). Hmm… what should I do? @ncw

**Comment by DWomack** [comment]
I am in middle of migrating my files from drive.filejump.com to the new filejump.com.  I have 380 gig on the first.  I am not a python programmer and having been using claude.ai to write and test code.  The code is finally running and 81 gig have been copied.  The standard python webDAV library wouldn't work with the new filejump.com. 

The ability to migrate files from drive.filejump.com to the new by using their cloud access interface didn't work.  You could browse your files and diretories on drive.filejump.com but when they transferred, they lost the file name.  It was a number.

**Comment by masrlinu** [comment]
WebDAV worked with my rclone app. You should get a VPS from [clouding.io](https://clouding.io) to transfer your files. They are cheap, offer fast transfer speeds, and if you search a bit, you might even find a €5 signup bonus for new customers ☺️

**Comment by DWomack** [comment]
I am going to try to use your code with rclone.  It is not in the current rclone.  Claude says I’ll have to build it.  ThanksSent from my iPhoneOn Nov 16, 2025, at 9:44 AM, masrlinu ***@***.***> wrote:﻿masrlinu left a comment (rclone/rclone#8114)
WebDAV worked with my rclone app. You should get a VPS from clouding.io to transfer your files. They are cheap, offer fast transfer speeds, and if you search a bit, you might even find a €5 signup bonus for new customers ☺️

**Comment by masrlinu** [comment]
No, it is not in the current build, because filejump changed their server. You can build my code with `go build` and then you can run it.

**Comment by DWomack** [comment]
I would like to run this on a raspberry pi that I have co located where there is symmetric fiber.

What version of go do I need?  Go is completely new to me. Claude will have to guide me

**Comment by masrlinu** [comment]
If you have Ubuntu or Debian, you can download it via
```
sudo apt-get install golang
```

and then in the folder of rclone
```
go build
```
and then rclone is compiled

**Comment by DWomack** [comment]
So on a pi that is local here go is installed:

go version go1.19.8 linux/arm64

I don't think that is the latest.  go1.25.4 linux/arm64 is the latest.

**Comment by masrlinu** [comment]
I don't know, if you need the latest version, you can try it also with the version which you have

**Comment by DWomack** [comment]
Sorry for being so dumb.  I downvoted the .zip from your GitHub.  So I unzip that into a directory. Change to that directory and rungo buildThat it?Sent from my iPhoneOn Nov 19, 2025, at 6:58 PM, masrlinu ***@***.***> wrote:﻿masrlinu left a comment (rclone/rclone#8114)
I don't know, if you need the latest version, you can try it also with the version which you have

**Comment by masrlinu** [comment]
Yes, did it work? Then you have an rclone program in your folder. This is the one with filejump support.

**Comment by DWomack** [comment]
The container did run.  And it successfully built rclone, but when I run rclone config, it doesn’t present any filejump options.  Looks like it is not including your PR stuff. I am going to try again on an amd64 box with docker on Ubuntu. I really like using a docker compose container to do these builds.  Makes it sorta portable.  I use a container called dockge to manage the rest of the containers on the box.  Really nice.  You just tell it to create a new container and paste the compose code in and click save.  Then click run.  It builds the alpine Linux instance with the required apps and go.  Grabs the stuff from GitHub and builds.  Couldn’t be easier. DennisSent from my iPhoneOn Nov 20, 2025, at 3:41 AM, masrlinu ***@***.***> wrote:﻿masrlinu left a comment (rclone/rclone#8114)
Yes, did it work? Then you have an rclone program in your folder. This is the one with filejump support.

**Comment by DWomack** [comment]
Got things working on two Pi and one ubuntu.

Pi uses this:
compose.yaml contains:
services:
  rclone-builder:
    image: golang:1.24-alpine
    container_name: rclone-builder
    working_dir: /build
    volumes:
      - ./rclone-bin:/output
    environment:
      - CGO_ENABLED=0
      - GOOS=linux
      - GOARCH=arm64
    command: sh -c "apk add --no-cache git make bash && git clone
      https://github.com/rclone/rclone.git . && git fetch origin
      pull/8693/head:pr-8693 && git checkout pr-8693 && make && cp rclone
      /output/ && /output/rclone version && /output/rclone help backends | grep
      -i filejump"
networks: {}

ubuntu uses this:
compose.yaml contains:
services:
  rclone-builder:
    image: golang:1.24-alpine
    container_name: rclone-builder
    working_dir: /build
    volumes:
      - ./rclone-bin:/output
    environment:
      - CGO_ENABLED=0
      - GOOS=linux
      - GOARCH=amd64
    command: sh -c "apk add --no-cache git make bash && git clone
      https://github.com/rclone/rclone.git . && git fetch origin
      pull/8693/head:pr-8693 && git checkout pr-8693 && make && cp rclone
      /output/ && /output/rclone version && /output/rclone help backends | grep
      -i filejump"
networks: {}

**Comment by masrlinu** [comment]
I answered you [here](https://github.com/rclone/rclone/pull/8693), you should read it. 

But I also want to add, that [NixOS](https://nixos.org/) is a Linux operating system and Nix is a programming language, which both belong together and make Docker obsolete. They are much much better and by the way NixOS had the biggest package manager in the world with more than 120.000 packages 😅 If you learn NixOS, you can play with your computer as much as you want and are able to completely revert your installed programs or your configuration. And you wouldn't need Docker!

**Comment by DWomack** [comment]
I have heard of NixOS. I just haven’t investigated it.  I’ll spin up a virtual and evaluate it.  That’s a euphemism for playing with it.  Thanks for the tip. Sent from my iPhoneOn Nov 21, 2025, at 8:22 PM, masrlinu ***@***.***> wrote:﻿masrlinu left a comment (rclone/rclone#8114)
I answered you here, you should read it.
But I also want to add, that NixOS is a Linux operating system and Nix is a programming language, which both belong together and make Docker obsolete. They are much much better and by the way NixOS had the biggest package manager in the world with more than 120.000 packages 😅 If you learn NixOS, you can play with your computer as much as you want and are able to completely revert your installed programs or your configuration. And you wouldn't need Docker!

**Comment by DWomack** [comment]
Just had an email from FileJump.  They have stopped the forced migration from drive.filejump.com to filejump.com.  I suspect they have encountered multilple issues with their implementation of WebDAV on filejump.com.  In my testing, I have found them.  Please push rclone to incorporate your drive.filejump backend into the offical rclone release.

Thanks

**Comment by masrlinu** [comment]
I don't know, how it will be long-term. They just stopped the forced migration until they find a stable solution. So my PR just helps until they migrate everyone.

**Comment by DWomack** [comment]
The new site filejump.com, has WebDAV.  Sorta.  The site is using php library by sabre From Claude…Yes, that’s right - it was Sabre/DAV that was causing the WebDAV issues. The filejump server you were working with had bugs in its Sabre/DAV implementation that were creating those 409 Conflict errors during your file migration work with rclone.The main problems you encountered were:	•	The server would return 409 errors when trying to create directories with certain characters or encodings in their names	•	It had trouble handling special characters in filenames properly	•	You ended up developing workarounds in your bash scripts to handle these server-side bugs, including strategies for dealing with problematic filenames and directory creation issuesSent from my iPhoneOn Dec 1, 2025, at 7:58 AM, masrlinu ***@***.***> wrote:﻿masrlinu left a comment (rclone/rclone#8114)
I don't know, how it will be long-term. They just stopped the forced migration until they find a stable solution. So my PR just helps until they migrate everyone.

**Comment by DWomack** [comment]
I found a WebDAV test.  It’s called litmus.  Check it out at GitHub.  It is from WebDAV.org. It does find problems with FileJump.comSent from my iPhoneOn Dec 1, 2025, at 1:45 PM, Dennis Womack ***@***.***> wrote:﻿The new site filejump.com, has WebDAV.  Sorta.  The site is using php library by sabre From Claude…Yes, that’s right - it was Sabre/DAV that was causing the WebDAV issues. The filejump server you were working with had bugs in its Sabre/DAV implementation that were creating those 409 Conflict errors during your file migration work with rclone.The main problems you encountered were:	•	The server would return 409 errors when trying to create directories with certain characters or encodings in their names	•	It had trouble handling special characters in filenames properly	•	You ended up developing workarounds in your bash scripts to handle these server-side bugs, including strategies for dealing with problematic filenames and directory creation issuesSent from my iPhoneOn Dec 1, 2025, at 7:58 AM, masrlinu ***@***.***> wrote:﻿masrlinu left a comment (rclone/rclone#8114)
I don't know, how it will be long-term. They just stopped the forced migration until they find a stable solution. So my PR just helps until they migrate everyone.

**Comment by DWomack** [comment]
Again, thank you for this work. What does it take to get your backend merged into rclone 1.73?I bought some more cloud storage from a company called Internxt.  They now have rclone support with a backend they wrote and maintain.  It is in rclone 1.73. I have a docker compose container that will do the build, if I knew exactly what to tell it for source. Thanks DennisSent from my iPhoneOn Dec 1, 2025, at 7:58 AM, masrlinu ***@***.***> wrote:﻿masrlinu left a comment (rclone/rclone#8114)
I don't know, how it will be long-term. They just stopped the forced migration until they find a stable solution. So my PR just helps until they migrate everyone.
